x-pack/test/functional/apps/monitoring/cluster/overview.js

```diff
@@ -159,5 +164,51 @@ export default function ({ getService, getPageObjects }) {
         expect(await overview.doesLsPanelExist()).to.be(false);
       });
     });
+
+    describe('Alerts', () => {
+      const { setup, tearDown } = getLifecycleMethods(getService, getPageObjects);
+
+      before(async () => {
+        await setup('x-pack/test/functional/es_archives/monitoring/singlecluster_green_gold', {
+          from: 'Aug 23, 2017 @ 21:29:35.267',
+          to: 'Aug 23, 2017 @ 21:47:25.556',
+        });
+      });
+
+      after(async () => {
+        await tearDown();
+
+        const apiResponse = await supertest.get('/api/alerts/_find?per_page=20');
+        const monitoringAlerts = apiResponse.body.data.filter(
+          ({ consumer }) => consumer === 'monitoring'
+        );
+
+        await Promise.all(
+          monitoringAlerts.map(async (alert) =>
+            supertest.delete(`/api/alerts/alert/${alert.id}`).set('kbn-xsrf', 'true').expect(204)
+          )
+        );
+
+        await browser.clearLocalStorage();
+      });
+
+      describe('when create alerts options is selected in the alerts modal', () => {
+        before(async () => {
+          await overview.acceptAlertsModal();
+        });
+
+        it('should show a toast when alerts are created successfully', async () => {
+          expect(await testSubjects.exists('alertsCreatedToast', { timeout: 10000 })).to.be(true);
+        });
+
+        it('should show badges when entering setup mode', async () => {
+          await setupMode.clickSetupModeBtn();
```

Comment:
Is it needed to leave setup mode once the test is done? Or does tearDown handle that?

Reply:
Good question! I don't think is needed because the setup state is stored in memory and in the URL as a query param and in the setup, we start with a fresh URL each time.
But, just in case this changes at some point, and to keep the test isolated I added a step to leave the setup mode.